metric_learn/mlkr.py

```diff
@@ -0,0 +1,124 @@
+"""
+Metric Learning for Kernel Regression (MLKR), Weinberger et al.,
+
+MLKR is an algorithm for supervised metric learning, which learns a distance
+function by directly minimising the leave-one-out regression error. This
+algorithm can also be viewed as a supervised variation of PCA and can be used
+for dimensionality reduction and high dimensional data visualization.
+"""
+from __future__ import division
+import numpy as np
+from six.moves import xrange
+
+from .base_metric import BaseMetricLearner
+
+class MLKR(BaseMetricLearner):
+    """Metric Learning for Kernel Regression (MLKR)"""
+    def __init__(self, A=None, epsilon=0.01):
+        """
+        MLKR initialization
+
+        Parameters
+        ----------
+        A: Initialization of matrix A. Defaults to the identity matrix.
+        epsilon: Step size for gradient descent
+        """
+        self.params = {
+            "A": A,
+            "epsilon": epsilon
+        }
+
+    def _process_inputs(self, X, y):
+        if X.ndim == 1:
```

Comment:
It's usually a good idea to convert inputs to numpy arrays here, just in case the user provided a plain Python list, or some other sequence.

```python
self.X = np.array(X, copy=False)
y = np.array(y, copy=False)
```